frontend/packages/shared/src/components/altinnHeader/AltinnHeader.test.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import { render as rtlRender, screen } from '@testing-library/react';
+import { AltinnHeader, AltinnHeaderProps } from './AltinnHeader';
+import {
+  ServicesContextProps,
+  ServicesContextProvider,
+} from 'app-development/common/ServiceContext';
+
+describe('AltinnHeader', () => {
+  it('should render AltinnHeaderMenu', () => {
+    render();
+    expect(screen.getByTestId('altinn-header-menu')).toBeInTheDocument();
+  });
+
+  it('should render AltinnHeaderButtons', () => {
+    render();
+    expect(screen.getByTestId('altinn-header-buttons')).toBeInTheDocument();
+  });
+
+  it('should render AltinnSubMenu when showSubMenu is true', () => {
+    render();
+    expect(screen.getByTestId('altinn-sub-menu')).toBeInTheDocument();
+  });
+
+  it('should not render AltinnSubMenu when showSubMenu is false', () => {
+    render({ showSubMenu: false });
+    expect(screen.getByTestId('altinn-sub-menu')).toBeInTheDocument();
+  });
+});
+
+const render = (props: Partial<AltinnHeaderProps> = {}) => {
+  const allProps = {
+    org: 'jest-test-org',
+    app: 'jest-test-app',
+    showSubMenu: true,
+  } as AltinnHeaderProps;
```

Comment:
Denne må ta inn `props`. `org` og `app` er heller ikke gyldige egenskaper her. Hvis du setter typen til `AltinnHeaderProps` i stedet for å caste, skal kodeverktøyet gi beskjed om feil som dette.
```suggestion
  const allProps: AltinnHeaderProps = {
    showSubMenu: true,
    ...props,
  };
```

Reply:
endret

Reply:
Ser at du fortsatt har type-casting (`... as AltinnHeaderProps`). Vi bør unngå det hvis det er andre måter å sette typen på. Her kan du sette typen direkte: `const allProps: AltinnHeaderProps = ...`